omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/program/ProgramHistoryFragmentController.java

```diff
@@ -75,8 +76,48 @@ public void controller(FragmentConfiguration config, @FragmentParam("app") AppDe
         List<PatientProgram> patientPrograms = programWorkflowService.getPatientPrograms(patientWrapper.getPatient(),
                 program, null, null, null, null, false);
 
-        // TODO: assumption here is that "getPatientPrograms" returns results in chronological order--need to confirm?
-        Collections.reverse(patientPrograms);
+        Collections.sort(patientPrograms, new Comparator<PatientProgram>() {
+            @Override
+            public int compare(PatientProgram patientProgram1, PatientProgram patientProgram2) {
+
+                if (patientProgram1.getDateEnrolled().after(patientProgram2.getDateEnrolled())) {
+                    return -1;
+                }
+                else if (patientProgram1.getDateEnrolled().before(patientProgram2.getDateEnrolled())) {
+                    return 1;
+                }
```

Comment:
For the above, could use OpenmrsUtil.compareWithNullAsEarliest(patientProgram1.getDateEnrolled(), patientProgram2.getDateEnrolled()), and return it if non-zero.